app/controllers/course/leaderboards_controller.rb

```diff
@@ -3,12 +3,17 @@ class Course::LeaderboardsController < Course::ComponentController
   before_action :check_component
   before_action :load_settings
   before_action :add_leaderboard_breadcrumb
+  before_action :check_group_leaderboard_settings, only: [:groups]
```

Comment:
This is the first method for disable-able components within a component, the other methods to check whether a component is enabled is named `check_component`. 

If we want to standardise, we can instead rename it `check_component_settings`, although that is not explicit for certain checks.